File: src/main/java/net/theevilreaper/vulpes/api/repository/AttributeRepository.java
package net.theevilreaper.vulpes.api.repository;

import io.micronaut.data.annotation.Repository;
import io.micronaut.data.repository.PageableRepository;
import net.theevilreaper.vulpes.api.model.AttributeEntity;

import java.util.UUID;

@Repository
public interface AttributeRepository extends PageableRepository<AttributeEntity, UUID> {
}

File: src/main/kotlin/net/theevilreaper/vulpes/api/repository/AttributeRepository.kt
package net.theevilreaper.vulpes.api.repository

import net.theevilreaper.vulpes.api.model.AttributeModel
import org.springframework.data.mongodb.repository.MongoRepository

/**
 * The repository is used to store different object instances from the [AttributeModel].
 * It extends the [MongoRepository] to provide the basic CRUD operations.
 * @see MongoRepository
 * @author theEvilReaper
 */
interface AttributeRepository : MongoRepository<AttributeModel, String>
